scipy/optimize/minpack.py

```diff
@@ -785,7 +785,10 @@ def curve_fit(f, xdata, ydata, p0=None, sigma=None, absolute_sigma=False,
         popt, pcov, infodict, errmsg, ier = res
         ysize = len(infodict['fvec'])
         cost = np.sum(infodict['fvec'] ** 2)
-        if ier not in [1, 2, 3, 4]:
+        if ier==5:
+            print("Optimal parameters not found: " + errmsg + " This can be increased by passing the parameter maxfev=n.")
```

Comment:
Just a quick note about two things:

1) Understandable that sometimes Python `warnings` don't show up, but perhaps that is something we can help get working. I just grepped through the code a bit and I do see `class OptimizeWarning(UserWarning)` defined and used through the optimize machinery. Perhaps one of the optimize regulars can comment on using that, along with a filter if one is needed.

2) Note that this part of the code is not covered by tests (codecov shows 0 % coverage on the diff). May be an opportunity to encode the desired warnings and exceptions situations in actual unit tests.

Reply:
Thanks for the help.

When I set `warnings.simplefilter('always')` just before `curve_fit` then it shows and another warning about not being able to find the covariance. I tried going through how warnings is set up but couldn't find anywhere where they're filtered yet.

I have also now added a test but don't have much experience of unit testing so not sure its right/tests for the right stuff. I don't know if I need to change something else too to add it properly, I don't really understand the workflow but happy to try and work it out more.